### PR TITLE
Adding Long Tail search support

### LINKED ISSUE #46: Support for institution search?
It appears there is no way to use this library to perform a search against Long Tail institutions. Was this excluded intentionally? If not, are there plans to support that? Is this something I would need to build myself?

### PR DESCRIPTION
**Pull request by Bpless**
Closes  #46

### CODE REVIEW

File: plaid/client.py [hunk 1]
```diff
@@ -455,3 +456,28 @@ def institution(self, url, institution_id):
...
             suppress_errors=self.suppress_http_errors
         )
+
+    @inject_url('institution_search')
+    def institution_search(self, url, q=None, p=None, institution_id=None):
+        '''
+        Perform simple search query against Long Tail institutions.
+
+        `q`   str    Query to match against the full list of institutions.
+        `p`    str   Filter FIs by a single product (Optional).
```
Comment (davidchambers) [inline plaid/client.py]: :triangular_ruler:

File: plaid/client.py [hunk 2]
```diff
@@ -455,3 +456,28 @@ def institution(self, url, institution_id):
...
+
+        assert q is not None or institution_id is not None, (
+            'query or institution_id required'
+        )
+
+        params = dict([(key, value) for key, value in [
+            ('q', q),
+            ('p', p),
+            ('id', institution_id)
+        ] if value is not None])
```
Comment (davidchambers) [inline plaid/client.py]: Could you use a dictionary comprehension here instead?

Comment (Bpless) [inline plaid/client.py]: I want to support Python 2.6 :(

Comment (davidchambers) [inline plaid/client.py]: If only there were a functional programming library in Python which supported filtering dictionaries. ;)

**Comment by davidchambers** [comment]
:deciduous_tree:

**Pull request merged**